src/core/toga/interface/widgets/switch.py

```diff
@@ -0,0 +1,102 @@
+from .base import Widget
+
+
+class Switch(Widget):
+    """
+    Switch widget, a clickable button with two stable states, True (on, checked) and False (off, unchecked)
+
+    :param label:       Text to be shown next to the Switch
+    :type label:        ``str``
+
+    :param id:          An identifier for this widget.
+    :type  id:          ``str``
+
+    :param style:       an optional style object. If no style is provided then a
+                        new one will be created for the widget.
+    :type style:        :class:`colosseum.CSSNode`
+
+    :param on_press:    Function to execute when pressed
+    :type on_press:     ``callable``
+
+    :param state        Current state of the Switch
+    :type state         ``Bool`
+    """
+
+    def __init__(self, label, id=None, style=None, on_press=None, state=False):
+        super().__init__(id=id, style=style, label=label, on_press=on_press, state=state)
+
+    def _configure(self, label, on_press, state):
+        self.label = label
+        self.on_press = on_press
+        self.state = state
```

Comment:
There should probably also be an "enabled/disabled" function; at which point, two new concerns arise:
* You could describe enabled/disabled as the "state" of the widget as well as the "on/off" state. 
* Is there a grammatical inconsistency in having "enabled" (which would suit an `if widget.enabled:` statement), as opposed to "state" (which lends itself it `if widget.stats is ON:` statements)? 

My immediate thought is that the `state` flag should be `on`, which would allow `if widget.on:`, but would also require `widget.on = False`. Any other suggestions?